src/wp-includes/class-wp-duotone.php

```diff
@@ -1180,9 +1180,14 @@ public static function output_footer_assets() {
 			echo self::get_svg_definitions( self::$used_svg_filter_data );
 		}
 
-		// This is for classic themes - in block themes, the CSS is added in the head via wp_add_inline_style in the wp_enqueue_scripts action.
-		if ( ! wp_is_block_theme() && ! empty( self::$used_global_styles_presets ) ) {
-			wp_add_inline_style( 'core-block-supports', self::get_global_styles_presets( self::$used_global_styles_presets ) );
+		// In block themes, the CSS is added in the head via wp_add_inline_style in the wp_enqueue_scripts action.
+		if ( ! wp_is_block_theme() ) {
+			if ( ! empty( self::$used_global_styles_presets ) ) {
+				wp_add_inline_style( 'core-block-supports', self::get_global_styles_presets( self::$used_global_styles_presets ) );
+			}
+			if ( ! empty( self::$block_css_declarations ) ) {
+				wp_add_inline_style( 'core-block-supports', wp_style_engine_get_stylesheet_from_css_rules( self::$block_css_declarations ) );
+			}
```

Comment:
```suggestion
			$handle = 'core-block-supports';
			wp_register_style( $handle, false );
			if ( ! empty( self::$used_global_styles_presets ) ) {
				wp_add_inline_style( $handle, self::get_global_styles_presets( self::$used_global_styles_presets ) );
			}
			if ( ! empty( self::$block_css_declarations ) ) {
				wp_add_inline_style( $handle, wp_style_engine_get_stylesheet_from_css_rules( self::$block_css_declarations ) );
			}
			wp_enqueue_style( $handle );
```

You can't add inline styles without registering and enqueuing style. 

This fixes the issue in my testing.

Reply:
I thought it was being registered already in script-loader, but it only looks like that happens when `wp_style_engine_get_stylesheet_from_context` returns something.

https://github.com/WordPress/wordpress-develop/blob/cdf3ac093de45c2c2d67fe0635f3d8a513a97895/src/wp-includes/script-loader.php#L3049-L3068

Would it be okay to always register `core-block-supports`?

Reply:
@ajlende Didn't know about that. You will need to enqueue it as well. Like this 

```
         wp_register_style( $style_tag_id, false );
	// Combines Core styles.
	if ( ! empty( $compiled_core_stylesheet ) ) {
		wp_add_inline_style( $style_tag_id, $compiled_core_stylesheet );
	}
	wp_enqueue_style( $style_tag_id );
```

Reply:
I'm just registered and enqueued a new stylesheet. The `'core-block-supports'` name is semi-dynamic, so we'll run into issues in the future if someone tries to add to `$core_styles_keys` and doesn't update duotone.

It's unfortunate that the duotone block supports styles won't be grouped with the rest of block supports, but it's probably not that big of a deal.